src/Stratis.Bitcoin/P2P/Peer/NetworkPeerConnection.cs

```diff
@@ -77,8 +77,14 @@ public IPEndPoint RemoteEndPoint
 
         /// <summary>Task completion that is completed when the work of the shutdown procedure is finished, including detaching from behaviors.</summary>
         /// <seealso cref="Shutdown"/>
+        /// <see cref="DisposeComplete"/>
         public TaskCompletionSource<bool> ShutdownComplete { get; private set; }
 
+        /// <summary>Task completion that is completed when the work of <see cref="Dispose"/> is finished.</summary>
+        /// <remarks>Note that first, <see cref="ShutdownComplete"/> is completed and then some more disposing 
+        /// occurs in <see cref="Dispose"/>. Only when all disposing work is done, this source is completed.</remarks>
+        public TaskCompletionSource<bool> DisposeComplete { get; private set; }
+
         /// <summary>Lock object to protect access to <see cref="shutdownInProgress"/>, <see cref="shutdownCalled"/>, <see cref="disposeRequested"/>, and <see cref="disposed"/> during the shutdown sequence.</summary>
         private object shutdownLock = new object();
```

Comment:
can be made readonly

Reply:
yeah, i'll make it readonly and move the initialization to the constructor